cli/template/base/src/env.mjs

```diff
@@ -31,45 +31,57 @@ const processEnv = {
 // Don't touch the part below
 // --------------------------
 
-const formatErrors = (
-  /** @type {z.ZodFormattedError<Map<string,string>,string>} */
-  errors,
-) =>
-  Object.entries(errors)
-    .map(([name, value]) => {
-      if (value && "_errors" in value)
-        return `${name}: ${value._errors.join(", ")}\n`;
-    })
-    .filter(Boolean);
+const merged = server.merge(client);
+/** @type z.infer<merged>
+ *  @ts-ignore - can't type this properly in jsdoc */
+let env;
 
-const isServer = typeof window === "undefined";
+if (!process.env.SKIP_ENV_VALIDATION) {
+  const formatErrors = (
+    /** @type {z.ZodFormattedError<Map<string,string>,string>} */
+    errors,
+  ) =>
+    Object.entries(errors)
+      .map(([name, value]) => {
+        if (value && "_errors" in value)
+          return `${name}: ${value._errors.join(", ")}\n`;
+      })
+      .filter(Boolean);
```

Comment:
i think we can remove this, the default `ZodError.flatten()` looks good

Reply:
<img width="656" alt="image" src="https://user-images.githubusercontent.com/8353666/217036139-37ce6cb5-22c5-4ac7-90fc-c293660b2aab.png">
looks nice